proposals/NNNN-reinstate-self-for-accessing-instance-properties-and-functions.md

```diff
@@ -0,0 +1,99 @@
+# Re-instate mandatory self for accessing instance properties and functions  
+
+* Proposal: [SE-NNNN](https://github.com/apple/swift-evolution/blob/master/proposals/NNNN-mandatory-self-for-accessing-instance-properties-and-functions.md)
+* Author(s): [David Hart](https://github.com/hartbit)
+* Status: **Review**
+* Review manager: TBD
+
+## Introduction
+
+Swift used to force using `self.` when accessing instance properties and functions, but since dropped this requirement (only keeping it for closure captures). It now feels inconsistent, and we have lost the automatic documentation of instance properties vs local variables and instance functions vs local functions or closures. This proposal offers to re-instate the original behaviour.
```

Comment:
Since the very first developer beta, there has never been a requirement to reference instance variables or methods with self outside of a closure. I understand this is just a point of history, but "re-instating" isn't quite the correct term for this proposal.